src/decision/location_selection.jl

```diff
@@ -0,0 +1,169 @@
+    containing ranks for single scenario.
+"""
+function site_selection(domain::Domain, scenario::DataFrameRow, w_scens::NamedDimsArray, dhw_scens::NamedDimsArray,
+    site_ids::Vector{Int64}, sum_cover::NamedDimsArray, area_to_seed::Float64)::Matrix{Int64}
+
+    mcda_vars = DMCDA_vars(domain, scenario, site_ids, sum_cover, area_to_seed, w_scens, dhw_scens)
+    n_sites = length(site_ids)
+
+    # site_id, seeding rank, shading rank
+    rankingsin = [mcda_vars.site_ids zeros(Int64, n_sites) zeros(Int64, n_sites)]
+
+    prefseedsites::Vector{Int64} = zeros(Int64, mcda_vars.n_site_int)
+    prefshadesites::Vector{Int64} = zeros(Int64, mcda_vars.n_site_int)
+
+    # Determine connectivity strength
+    # Account for cases where no coral cover
+    in_conn, out_conn, strong_pred = connectivity_strength(domain.TP_data .* site_k_area(domain), Array(sum_cover))
+    in_conn = in_conn[site_ids]
+    out_conn = out_conn[site_ids]
+    strong_pred = strong_pred[site_ids]
+
+    (_, _, ranks) = guided_site_selection(mcda_vars, scenario.guided, true, true, prefseedsites, prefshadesites, rankingsin, in_conn, out_conn, strong_pred)
+
+    return ranks[:, 2:3]
+end
+
+"""
+    run_site_selection(domain::Domain, scenarios::DataFrame, sum_cover::AbstractArray, area_to_seed::Float64; aggregate_function=nothing,
+        target_seed_sites=nothing, target_shade_sites=nothing)
+
+Perform site selection for a given domain for multiple scenarios defined in a dataframe.
+
+# Arguments
+- `domain` : ADRIA Domain type, indicating geographical domain to perform site selection over.
+- `scenarios` : DataFrame of criteria weightings and thresholds for each scenario.
+- `sum_cover` : array of size (number of scenarios * number of sites) containing the summed coral cover for each site selection scenario.
+- `area_to_seed` : area of coral to be seeded at each time step in km^2
+- `aggregate_function` : function which aggregates ranks output into preferred form, e.g `ranks_to_frequencies`, `ranks_to_location_order`.
+- `target_seed_sites` : list of candidate locations for seeding (indices)
+- `target_shade_sites` : list of candidate location to shade (indices)
+
+# Returns
+- `ranks_store` : number of scenarios * sites * 3 (last dimension indicates: site_id, seed rank, shade rank)
+    containing ranks for each scenario run.
+"""
+function run_site_selection(domain::Domain, scenarios::DataFrame, sum_cover::AbstractArray, area_to_seed::Float64; aggregate_function=nothing,
+    target_seed_sites=nothing, target_shade_sites=nothing)
+    ranks_store = NamedDimsArray(
+        zeros(length(domain.site_ids), 2, nrow(scenarios)),
+        sites=domain.site_ids,
+        intervention=1:2,
+        scenarios=1:nrow(scenarios),
+    )
+
+    dhw_scens = domain.dhw_scens
+    wave_scens = domain.wave_scens
+
+    # Pre-calculate maximum depth to consider
+    scenarios[:, "max_depth"] .= scenarios.depth_min .+ scenarios.depth_offset
```

Comment:
Avoid modifying the user provided dataframe. You don't need `max_depth` as a column anyway